boxoffice/views/order.py

```diff
@@ -0,0 +1,198 @@
+
+        line_item = LineItem(order=order, item=item, discount_policy=policy,
+            line_item_no=idx+1,
+            discount_coupon=coupon,
+            ordered_at=datetime.utcnow(),
+            base_amount=line_item_tup.base_amount,
+            discounted_amount=line_item_tup.discounted_amount,
+            final_amount=line_item_tup.base_amount-line_item_tup.discounted_amount)
+        if coupon:
+            coupon.register_use()
+            db.session.add(coupon)
+        db.session.add(line_item)
+
+    db.session.add(order)
+    db.session.commit()
+    return jsonify(code=200, order_id=order.id,
+        order_access_token=order.access_token,
+        payment_url=url_for('payment', order=order.id),
+        free_order_url=url_for('free', order=order.id),
+        final_amount=order.get_amounts().final_amount)
+
+
+@app.route('/<order>/free', methods=['GET', 'OPTIONS', 'POST'])
+@load_models(
+    (Order, {'id': 'order'}, 'order')
+    )
+@xhr_only
+@cross_origin(origins=ALLOWED_ORIGINS)
+def free(order):
+    """
+    Completes a order which has a final_amount of 0
+    """
+    order_amounts = order.get_amounts()
+    if order_amounts.final_amount == 0:
+        order.confirm_sale()
+        db.session.add(order)
+        db.session.commit()
+        return jsonify(code=200)
+    else:
+        return api_result(402, 'Free order confirmation failed')
+
+
+@app.route('/<order>/payment', methods=['GET', 'OPTIONS', 'POST'])
+@load_models(
+    (Order, {'id': 'order'}, 'order')
+    )
+@xhr_only
+@cross_origin(origins=ALLOWED_ORIGINS)
+def payment(order):
+    """
+    Accepts JSON containing `pg_payment_id`.
+
+    Creates a payment object, attempts to 'capture' the payment from Razorpay,
+    and returns a JSON containing the result of the operation.
+
+    A successful capture results in a `payment_transaction` registered against the order.
+    """
+    pg_payment_id = request.json.get('pg_payment_id')
+    if not (request.json and pg_payment_id):
+        abort(400)
+
+    order_amounts = order.get_amounts()
+    online_payment = OnlinePayment(pg_payment_id=pg_payment_id, order=order)
+
+    rp_resp = razorpay.capture_payment(online_payment.pg_payment_id, order_amounts.final_amount)
+    if rp_resp.status_code == 200:
+        online_payment.confirm()
+        db.session.add(online_payment)
+        transaction = PaymentTransaction(order=order, online_payment=online_payment, amount=order_amounts.final_amount)
+        db.session.add(transaction)
+        order.confirm_sale()
+        db.session.add(order)
+        db.session.commit()
+        boxofficeq.enqueue(send_invoice_email, order.id)
+        return jsonify(code=200)
+    else:
+        online_payment.fail()
+        db.session.commit()
+        raise PaymentFailError("Online payment failed for order - {order}".format(order=order.id))
+        return api_result(402, 'Payment capture failed')
+
+
+@app.route('/<access_token>/receipt', methods=['GET'])
```

Comment:
Take this off the top-level and change to something like `/order/<access_token>`